R/data-model-helpers.R

```diff
@@ -245,3 +245,50 @@ data_model_db_types_to_R_types <- function(data_model) {
   data_model$columns$type <- new_type
   data_model
 }
+
+upd_pks_after_rename <- function(
+  pks,
+  table_name,
+  list_of_renames) {
+
+  pk_of_table <- filter(pks, table == !!table_name) %>% pull(column)
+  if (any(list_of_renames == pk_of_table)) {
+    new_pk <- names(list_of_renames[list_of_renames == pk_of_table])
+    pks$column[pks$table == table_name] <- new_pk
+  }
+  pks
+}
+
+upd_fks_after_rename <- function(
```

Comment:
This is where I think a longer-form data structure with three columns (table, column, role) would fare much better: we'd only need one update routine. TBD.